File: 2024-11-November-GFG-POTD/November 15 - Second Largest/Solution.py
class Solution:
    def getSecondLargest(self, arr):
        if len(arr) < 2:
            return -1  # Return -1 if there are not enough elements for a second largest
        
        largest = float('-inf')
        second_largest = float('-inf')

        for num in arr:
            if num > largest:
                second_largest = largest  # update second largest before updating largest
                largest = num
            elif num > second_largest and num < largest:
                second_largest = num  # update second largest if num is less than largest but greater than second_largest

        return second_largest if second_largest != float('-inf') else -1


if __name__ == "__main__":
    t = int(input().strip())
    
    for _ in range(t):
        arr = list(map(int, input().strip().split()))
        ob = Solution()
        ans = ob.getSecondLargest(arr)
        print(ans)
        print("~")

File: 2024-11-November-GFG-POTD/November 15 - Second Largest/Solution.java
//{ Driver Code Starts
// Initial Template for Java
import java.io.*;
import java.util.*;
import java.util.stream.Collectors;

public class Main {

    public static void main(String[] args) throws Exception {
        Scanner sc = new Scanner(System.in);
        int t = Integer.parseInt(sc.nextLine());
        while (t-- > 0) {
            String[] arr1Str = sc.nextLine().split(" ");
            int[] arr = Arrays.stream(arr1Str).mapToInt(Integer::parseInt).toArray();
            Solution ob = new Solution();
            int ans = ob.getSecondLargest(arr);
            System.out.println(ans);

            System.out.println("~");
        }
    }
}

// } Driver Code Ends


// User function Template for Java

class Solution {
    public int getSecondLargest(int[] arr) {
        // Code Here
         if (arr.length < 2) {
            return -1; // Return -1 if there are not enough elements for a second largest
        }
        
        int largest = Integer.MIN_VALUE;
        int secondLargest = Integer.MIN_VALUE;

        for (int num : arr) {
            if (num > largest) {
                secondLargest = largest;  // update second largest before updating largest
                largest = num;
            } else if (num > secondLargest && num < largest) {
                secondLargest = num;  // update second largest if num is less than largest but greater than secondLargest
            }
        }

        return (secondLargest == Integer.MIN_VALUE) ? -1 : secondLargest;
    }
}
